cfgmgr/macsecmgr.cpp

```diff
@@ -78,6 +78,46 @@ static void lexical_convert(const std::string &cipher_str, MACsecMgr::MACsecProf
     }
 }
 
+
+/* Decodes a Type 7 encoded input.
+ *
+ * The Type 7 encoded string consists of two decimal digits, followed by a series of hexadecimal characters
+ *
+ * An example: encoding of string "password" is 044B0A151C36435C0D.
+ * This has a salt/offset of 4 (04 in the example), and the encoded string for "password" is 4B0A151C36435C0D.
+ *
+ * The algorithm is a straightforward XOR Cipher that relies on an ascii-encoded 53-byte magic salt string
+ *
+ * Decode()
+ *    Get the salt index from the first 2 chars
+ *    For each byte in the provided text after the encoded salt:
+ *        j = (salt index + 1) % 53
+ *        XOR the i'th byte of the password with the j'th byte of the magic constant.
+ *        append to the decoded string.
+ */
+static std::string decodeKey(const std::string &cipher_str)
```

Comment:
I feel we should add the zero padding if cipher_str doesn't meet the requirement(not 32/64 bytes length). Otherwise, it will crash if the length of cipher_str is an odd number.

Reply:
The 32/64 byte is for the plain text CAK string which we were using earlier -- that was not type7 encoded. 

Here the input **cipher_str** is the encoded string which can be either of length 66 bytes or 130 bytes. This will have following properties. 

1. 66 bytes of length, for 128-byte cipher suite
           > first 2 bytes of the string will be the index from the magic salt string. 
           > remaining 64 bytes will be encoded string from the 32-byte plain text CAK. 

2. 130 bytes of length, for 256-byte cipher suite
           > first 2 bytes of the string will be the index from the magic salt string. 
           > remaining 128 bytes will be encoded string from the 32 byte plain text CAK

I will add a check to make sure the length is either of the above 66/130 bytes -- if not will raise and Error log and not proceed further.